File: 30-Days-of-Code/Day-2-Operators/Go/solution.go
package main

import (
    "fmt"
)

func main() {
           
    var mealCost float32
    var tipPercent uint32
    var taxPercent uint32
    
    fmt.Scanf("%f", &mealCost)
    fmt.Scanf("%d", &tipPercent)
    fmt.Scanf("%d", &taxPercent)
    
    var tip float32 = mealCost * (float32(tipPercent) / 100.0)
    var tax float32 = mealCost * (float32(taxPercent) / 100.0)
    
    var totalCost float32 = mealCost + tip + tax
    
    var roundedTotalCost uint32 = uint32(totalCost + 0.5)
    
    fmt.Printf("The total meal cost is %d dollars.\n", roundedTotalCost)

}

File: 30-Days-of-Code/Day-2-Operators/Pascal/solution.pas
uses sysutils, math, crt;

var mealCost : double;
var tipPercent : byte;
var taxPercent : byte;
var tip : double;
var tax : double;
var totalCost : double;
var roundedTotalCost : byte;

begin
    mealCost := 0;
    tipPercent := 0;
    taxPercent := 0;
    readln(mealCost);
    readln(tipPercent);
    readln(taxPercent);
    
    tip := mealCost * (tipPercent / 100);
    tax := mealCost * (taxPercent / 100);
    totalCost := mealCost + tip + tax;
    
    roundedTotalCost := floor(totalCost + 0.5);
    
    writeln('The total meal cost is ', inttostr(roundedTotalCost), ' dollars.');
    
end.
